src/Crawler.php

```diff
@@ -490,21 +518,21 @@ public function addToCrawlQueue(CrawlUrl $crawlUrl): self
             return $this;
         }
 
-        $this->crawledUrlCount++;
-
         $this->crawlQueue->add($crawlUrl);
 
         return $this;
     }
 
-    public function maximumCrawlCountReached(): bool
+    public function reachedCrawlLimits(): bool
     {
-        $maximumCrawlCount = $this->getMaximumCrawlCount();
-
-        if (is_null($maximumCrawlCount)) {
-            return false;
-        }
+        $totalCrawlLimit = $this->getTotalCrawlLimit();
+        $currentCrawlLimit = $this->getCurrentCrawlLimit();
 
-        return $this->getCrawlerUrlCount() >= $maximumCrawlCount;
+        return (
```

Comment:
This is hard to read and hard to debug. Use early returns as much as possible here.

Reply:
Done @freekmurze